src/gcp/location.ts

```diff
@@ -0,0 +1,100 @@
+// A flattening of container_registry_hosts and
+// region_multiregion_map from regionconfig.borg
+const SUBDOMAIN_MAPPING: Record<string, string> = {
+  "us-west1": "us",
+  "us-west2": "us",
+  "us-west3": "us",
+  "us-west4": "us",
+  "us-central1": "us",
+  "us-central2": "us",
+  "us-east1": "us",
+  "us-east4": "us",
+  "northamerica-northeast1": "us",
+  "southamerica-east1": "us",
+  "europe-west1": "eu",
+  "europe-west2": "eu",
+  "europe-west3": "eu",
+  "europe-west4": "eu",
+  "europe-west5": "eu",
+  "europe-west6": "eu",
+  "europe-central2": "eu",
+  "europe-north1": "eu",
+  "asia-east1": "asia",
+  "asia-east2": "asia",
+  "asia-northeast1": "asia",
+  "asia-northeast2": "asia",
+  "asia-northeast3": "asia",
+  "asia-south1": "asia",
+  "asia-southeast2": "asia",
+  "australia-southeast1": "asia",
+};
+
+/** Google Cloud Storage multi-region mapping (https://cloud.google.com/storage/docs/locations#location-mr) */
+const MULTI_REGION_MAPPING: Record<string, string> = {
+  /** us */
+  "us-central1": "us",
+  "us-east1": "us",
+  "us-east4": "us",
+  "us-west1": "us",
+  "us-west2": "us",
+  "us-west3": "us",
+  "us-west4": "us",
+  /** eu */
+  "europe-central2": "eu",
+  "europe-north1": "eu",
+  "europe-west1": "eu",
+  "europe-west3": "eu",
+  "europe-west4": "eu",
+  "europe-west5": "eu",
+  /** asia */
+  "asia-east1": "asia",
+  "asia-east2": "asia",
+  "asia-northeast1": "asia",
+  "asia-northeast2": "asia",
+  "asia-northeast3": "asia",
+  "asia-south1": "asia",
+  "asia-south2": "asia",
+  "asia-southeast1": "asia",
+  "asia-southeast2": "asia",
+};
+
+/** Google Cloud Storage dual-region mapping (https://cloud.google.com/storage/docs/locations#location-dr) */
+const DUAL_REGION_MAPPING: Record<string, string> = {
+  /** asia1 */
+  "asia-northeast1": "asia1",
+  "asia-northeast2": "asia1",
+  /** eur4 */
+  "europe-north1": "eur4",
+  "europe-west4": "eur4",
+  /** nam4 */
+  "us-central1": "nam4",
+  "us-east1": "nam4",
+};
+
+/**
+ * Helper function to return a mapping from a geographical region to a subdomain.
+ * This mapping includes subdomains that can be outside the geographical multi-region mapping
+ * Some examples are 'southamerica-east1' -> 'us' or 'australia-southeast1' -> 'asia'
+ * @returns a Record<string, string> of region to subdomain
+ */
+export function getSubdomainMapping(): Record<string, string> {
+  return SUBDOMAIN_MAPPING;
+}
```

Comment:
I was a bit hesitant since we'd have to expose api's that basically re-implement the underlying Record features, like Object.keys() & Object.values(). But if we ever end up changing the data structure, it would be nice to have this interface. Ended up adding `regionToSubdomain()`, `getContainerRegistryRegions()`, and `getContainerRegistrySubdomains()` in the latest commit.